osu.Game/Rulesets/Mods/Mod.cs

```diff
@@ -105,32 +105,45 @@ public abstract class Mod : IMod, IEquatable<Mod>, IDeepCloneable<Mod>
         /// </summary>
         /// <example>
         /// <list type="bullet">
-        /// <item><see cref="ModHardRock"/> is valid for multiplayer.</item>
+        /// <item><see cref="ModHardRock"/> <b>is</b> valid for multiplayer.</item>
         /// <item>
-        /// <see cref="ModDoubleTime"/> is valid for multiplayer as long as it is a <b>required</b> mod,
+        /// <see cref="ModDoubleTime"/> <b>is</b> valid for multiplayer as long as it is a <b>required</b> mod,
         /// as that ensures the same duration of gameplay for all users in the room.
         /// </item>
         /// <item>
-        /// <see cref="ModAdaptiveSpeed"/> is not valid for multiplayer, as it leads to varying
+        /// <see cref="ModAdaptiveSpeed"/> <b>is not</b> valid for multiplayer, as it leads to varying
         /// gameplay duration depending on how the users in the room play.
         /// </item>
-        /// <item><see cref="ModAutoplay"/> is not valid for multiplayer.</item>
+        /// <item><see cref="ModAutoplay"/> <b>is not</b> valid for multiplayer.</item>
         /// </list>
         /// </example>
         [JsonIgnore]
         public virtual bool ValidForMultiplayer => true;
 
+        /// <summary>
+        /// Whether this mod can be specified as a "required" mod on freestyle playlist items, indicating that all rulesets contain an implementation of this mod.
+        /// </summary>
+        /// <example>
+        /// <list type="bullet">
+        /// <item><see cref="ModHardRock"/> <b>is</b> valid as a freestyle required-mod.</item>
+        /// <item>
+        /// <c>OsuModNoScope</c> <b>is not</b> valid, as it is only implemented in the osu! ruleset.
+        /// </item>
+        /// </list>
+        /// </example>
+        public virtual bool ValidForFreestyleAsRequiredMod => false;
```

Comment:
First of all, this should exist on `IMod` too.

Secondly, these flags are getting a bit much. Maybe not in this PR, but I'd think about doing some sort of a consolidation pass on these to make getting through these easier. Something like (zero idea if viable, just spitballing):

```
Mod
  Valid
      SoloGameplay: bool // replaces UserPlayable
      Multiplayer
          [Flags] RequiredMod: Freestyle | NotFreestyle // combines ValidForMultiplayer and ValidForFreestyleAsRequiredMod
          [Flags] AllowedMod: WhenNotSpecified | WhenSpecified // combines ValidForMultiplayerAsFreeMod and AlwaysValidForSubmission
```